src/panels/lovelace/strategies/areas/editor/ha-items-visibility-editor.ts

```diff
@@ -0,0 +1,232 @@
+import { mdiDrag, mdiEye, mdiEyeOff } from "@mdi/js";
+import { LitElement, css, html, nothing } from "lit";
+import { customElement, property } from "lit/decorators";
+import { classMap } from "lit/directives/class-map";
+import { repeat } from "lit/directives/repeat";
+import memoizeOne from "memoize-one";
+import { fireEvent } from "../../../../../common/dom/fire_event";
+import "../../../../../components/ha-icon";
+import "../../../../../components/ha-icon-button";
+import "../../../../../components/ha-icon-button-next";
+import "../../../../../components/ha-md-list";
+import "../../../../../components/ha-md-list-item";
+import "../../../../../components/ha-sortable";
+import "../../../../../components/ha-svg-icon";
+import type { HomeAssistant } from "../../../../../types";
+
+export interface VisibilityEditorItem {
+  icon?: string;
+  iconPath?: string;
+  value: string;
+  label: string;
+  description?: string;
+}
+
+export interface VisibilityEditorValue {
+  order: string[];
+  hidden: string[];
+}
+
+declare global {
+  interface HTMLElementTagNameMap {
+    "ha-items-visibility-editor": HaItemVisibilityEditor;
+  }
+  interface HASSDomEvents {
+    "item-navigate-clicked": { value: string };
+  }
+}
+
+@customElement("ha-items-visibility-editor")
```

Comment:
We now have `dialog-area-filter` and this new element, they seem to do pretty much the same thing, is the plan so merge the 2 at some point? Or replace `dialog-area-filter` with this new element?